packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -268,20 +260,46 @@ export function markAllThreadsInTeamRead(userId: string, teamId: string) {
     };
 }
 
-export function markThreadAsUnread(userId: string, teamId: string, threadId: string, postId: string) {
+export function markThreadAsUnread(userId: string, teamId: string, threadId: string) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        try {
-            await Client4.markThreadAsUnreadForUser(userId, teamId, threadId, postId);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch, getState);
-            dispatch(logError(error));
-            return {error};
+        const getPostsForThread = makeGetPostsForThread();
+        let posts = getPostsForThread(getState(), threadId);
+
+        // load posts in thread if they are not loaded already
+        if (posts.length < 2) {
+            getPostThread(threadId)(dispatch, getState).then(({data, error}) => {
+                if (data) {
+                    posts = getPostsForThread(getState(), threadId);
+                    markThreadAsUnreadForUser(dispatch, getState, {userId, teamId, threadId, lastPostId: posts[0].id});
+                } else if (error) {
+                    return {error};
+                }
+                return {};
+            });
+        } else {
+            markThreadAsUnreadForUser(dispatch, getState, {userId, teamId, threadId, lastPostId: posts[0].id});
         }
 
         return {};
     };
 }
 
+function markThreadAsUnreadForUser(
+    dispatch: DispatchFunc,
+    getState: GetStateFunc,
+    params: {userId: string; teamId: string; threadId: string; lastPostId: string},
+) {
```

Comment:
since we use `redux-thunk` you don't need to explicitly pass `dispatch`, and `getState`,
you can just:

```suggestion
function markThreadAsUnreadForUser(

    params: {userId: string; teamId: string; threadId: string; lastPostId: string},
) {
    return (dispatch, getState) => {
```

Reply:
Great suggestion, thanks 👍